Minimum node version is too low for copilot to run

### Summary

I am running a local build of zed (b158ee60ca) on Ubuntu 24.04.2 LTS.

I get an error that copilot cannot be installed:

![Image](https://github.com/user-attachments/assets/cf5e772b-42c2-4f30-afbb-27d031cfad65)

In the zed logs I see:

```
2025-04-02T13:38:20.397462938+02:00 [INFO] starting language server process. binary path: "/usr/bin/node", working directory: "/", args: ["/home/kroeger/.local/share/zed/copilot/node_modules/@github/copilot-language-server/dist/language-server.js", "--stdio"]
2025-04-02T13:38:20.451653388+02:00 [ERROR] cannot read LSP message headers
2025-04-02T13:38:20.451725411+02:00 [ERROR] Broken pipe (os error 32)
2025-04-02T13:38:20.451751403+02:00 [ERROR] server shut down
```

My node version of `/usr/bin/node` is `v18.19.1`.

If I execute the command from the log myself, I get:

```
/usr/bin/node /home/kroeger/.local/share/zed/copilot/node_modules/@github/copilot-language-server/dist/language-server.js --stdio

Node.js 20.x is required to run GitHub Copilot but found 18.19.1
```

When I change

```patch
diff --git i/crates/node_runtime/src/node_runtime.rs w/crates/node_runtime/src/node_runtime.rs
index 80d7da1373..c3667a93c7 100644
--- i/crates/node_runtime/src/node_runtime.rs
+++ w/crates/node_runtime/src/node_runtime.rs
@@ -473,7 +473,7 @@ pub struct SystemNodeRuntime {
 }
 
 impl SystemNodeRuntime {
-    const MIN_VERSION: semver::Version = Version::new(18, 0, 0);
+    const MIN_VERSION: semver::Version = Version::new(20, 0, 0);
     async fn new(node: PathBuf, npm: PathBuf) -> Result<Box<dyn NodeRuntimeTrait>> {
         let output = util::command::new_smol_command(&node)
             .arg("--version")
```

zed downloads a newer node for me, and with that, it works 🎉 

```
2025-04-02T14:11:14.350991798+02:00 [INFO] starting language server process. binary path: "/home/kroeger/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/", args: ["/home/kroeger/.local/share/zed/copilot/node_modules/@github/copilot-language-server/dist/language-server.js", "--stdio"]
2025-04-02T14:11:14.529359291+02:00 [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.294.0 initialized"
}
2025-04-02T14:11:14.529406797+02:00 [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
```

### Zed Version and System Specs

Zed: v0.181.0 (Zed Dev b158ee60cacc1e34c0244206938cd2232b28138f) 
OS: Linux Wayland ubuntu 24.04
Memory: 62.5 GiB
Architecture: x86_64
GPU: NVIDIA RTX 2000 Ada Generation Laptop GPU || NVIDIA || 550.120

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -473,7 +473,7 @@ pub struct SystemNodeRuntime {
 }
 
 impl SystemNodeRuntime {
-    const MIN_VERSION: semver::Version = Version::new(18, 0, 0);
+    const MIN_VERSION: semver::Version = Version::new(20, 0, 0);
     async fn new(node: PathBuf, npm: PathBuf) -> Result<Box<dyn NodeRuntimeTrait>> {
         let output = util::command::new_smol_command(&node)
             .arg("--version")

